### PR TITLE
chore: Unit-test framework migration to testing-library

### PR DESCRIPTION
**Pull request by maraisr**
With this PR I aim to remove enzyme completely from the project to enable more meaningful tests.

### CODE REVIEW

**Comment by txwkx** [comment]
_Casually_ leaving a comment to get notified when this PR would be merged. Looking forward to improving the coverage and unit tests of my companies OSS component library.

@maraisr saw your tweet, good job on the migration 🐐, keep it up!

File: lib/components/TextInput/TextInput.spec.jsx [hunk 1]
```diff
@@ -1,167 +1,160 @@
...
+		const { container } = render(
 			<TextInput
-				className="text-input-class"
+				className="input-class"
 				placeholder="placeholder something"
 			/>,
 		);
-		expect(textInput.find('input').prop('type')).toEqual('text');
-		textInput.unmount();
+		expect(container.querySelector('input')).toBeInTheDocument();
```
Comment (amir-zahedi) [inline lib/components/TextInput/TextInput.spec.jsx]: How are we testing the input type here?

File: lib/components/Typography/Text/Text.spec.jsx [hunk 2]
```diff
@@ -1,27 +1,30 @@
...
 
 	describe('when muted', () => {
 		it('should match snapshot', () => {
-			expect(render(<Text muted>Hello World</Text>)).toMatchSnapshot();
+			expect(
+				render(<Text muted>Hello World</Text>).container.firstChild,
+			).toMatchSnapshot();
 		});
 	});
 });
```
Comment (amir-zahedi) [inline lib/components/Typography/Text/Text.spec.jsx]: Guess we should add some tests for text sizes as well as component it renders at some point. Not necessarily in this PR though

File: lib/components/ModalBase/ModalPortal.spec.jsx [hunk 3]
```diff
@@ -1,292 +1,152 @@
...
 	describe('when portal', () => {
 		it('should be added when open', () => {
-			expect(document.body.innerHTML).toBe('');
-
-			const modal = mount(
-				<NakedModal isOpen>
-					<p>Hello World!</p>
-				</NakedModal>,
+			const { getByRole } = render(
+				<NakedModal isOpen>Hello World!</NakedModal>,
```
Comment (amir-zahedi) [inline lib/components/ModalBase/ModalPortal.spec.jsx]: Same here

File: lib/components/ModalBase/ModalPortal.spec.jsx [hunk 4]
```diff
@@ -1,292 +1,152 @@
...
+			expect(getByRole('dialog')).not.toHaveTextContent('Hello World!');
 		});
 
 		it('should add children when open', () => {
-			const modal = mount(
-				<NakedModal isOpen>
-					<p>Hello World!</p>
-				</NakedModal>,
+			const { getByRole } = render(
+				<NakedModal isOpen>Hello World!</NakedModal>,
```
Comment (amir-zahedi) [inline lib/components/ModalBase/ModalPortal.spec.jsx]: And here unless I'm missing something

File: lib/components/InputBase/NotchedBase.spec.jsx [hunk 5]
```diff
@@ -1,124 +1,112 @@
...
-				isActive
-				className="notched-class"
-				placeholder="placeholder something"
-			/>,
-		);
-		expect(notch.html()).toContain('notchedBaseActive');
-		notch.unmount();
-	});
-
-	it('should not add dirty style to dom element when isEmpty prop is not true', () => {
```
[the remark this answers is not in the document]
Comment (maraisr) [reply lib/components/InputBase/NotchedBase.spec.jsx]: There is no `isDirty` prop for this component 🤷🏼‍♂️

File: lib/components/LinearProgressIndicator/LinearProgressIndicator.spec.jsx [hunk 6]
```diff
@@ -1,51 +1,14 @@
...
 describe('<LinearProgressIndicator />', () => {
 	it('should not throw', () =>
-		expect(() => shallow(<LinearProgressIndicator />)).not.toThrow());
+		expect(() => render(<LinearProgressIndicator />)).not.toThrow());
 
 	it('should match snapshot', () => {
-		expect(shallow(<LinearProgressIndicator />)).toMatchSnapshot();
-	});
-
-	it('should add a root div element', () => {
```
[the remark this answers is not in the document]
Comment (maraisr) [reply lib/components/LinearProgressIndicator/LinearProgressIndicator.spec.jsx]: If we think about it - really want this test is doing is saying "there needs to be a div with root class"... Well duh 🤦‍♂️ that is what the component renders. Seeing as there is no _logic_ on this component, I feel this is purely a visual component, so Chromatic confirms that + the snapshot houses the dom tests. So Im quite comfortable that between those 2 tests, this component is covered.

File: lib/components/Meta/Meta.spec.jsx [hunk 7]
```diff
@@ -9,66 +9,37 @@ const TestIcon = () => (
...
-	});
-
-	it('should pass on className to dom element', () => {
-		const meta = shallow(
-			<Meta className="meta-class" label="Hello World!" />,
-		);
-		expect(meta.find('span').hasClass('meta-class')).toBeTruthy();
-	});
-
-	it('should by default apply Primary variant styles', () => {
```
[the remark this answers is not in the document]
Comment (maraisr) [reply lib/components/Meta/Meta.spec.jsx]: Ah - good point.

File: lib/components/ModalBase/ModalPortal.spec.jsx [hunk 8]
```diff
@@ -1,292 +1,152 @@
...
+			render(
+				<NakedModal isOpen={false}>
+					<p>Hello, I am a modal body!</p>
+				</NakedModal>,
+			),
+		).not.toThrow();
+	});
 
-		expect(() => mount(modal).unmount()).not.toThrow();
+	it('should not throw when open', () => {
```
[the remark this answers is not in the document]
Comment (maraisr) [reply lib/components/ModalBase/ModalPortal.spec.jsx]: No - boolean props are true just by being defined. eg: `<Element isOpen/> === <Element isOpen={true}/>` and `<Element /> === <Element isOpen={false}/>`.

File: lib/components/Radio/Radio.spec.jsx [hunk 9]
```diff
@@ -49,21 +51,11 @@ describe('<RadioButton />', () => {
 							checked={false}
 						/>
 					</RadioGroup>,
-				),
-			).toMatchSnapshot());
-
-		it('should match the snapshot for a focused radio', () =>
```
[the remark this answers is not in the document]
Comment (maraisr) [reply lib/components/Radio/Radio.spec.jsx]: Focus is purely handled with css - and jest doesn't touch our css at all. So pointless test.

File: lib/components/TextInput/TextInput.spec.jsx (hunk 1 above)
Comment (maraisr) [reply lib/components/TextInput/TextInput.spec.jsx]: 🤦‍♂️ touche

**Pull request merged**